Manual backport: doc: Updated ABI compatibility Ironside ver (#30036)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/abi_compatibility.rst

The following sections provide detailed lists of changes by component.

IronSide Secure Enclave (IronSide SE) v23.7.0+30
================================================

Added
-----

> * Added ``ironside_se_mpcconf_radioram_access()`` to the MPCCONF service API, enabling access to the radio domain RAM from the application core for bootloader purposes.

Reviewer: ```suggestion
* Added ``ironside_se_mpcconf_radioram_access()`` to the MPCCONF service API.
  The function enables access to the radio domain RAM from the application core for bootloader purposes.
```

Updated
-------

> * IronSide SE resilience against MRAM corruption is improved, as secure domain CPU bus faults now trigger snapshot recovery. (NCSDK-37077)
>   Corruption in other file systems used by IronSide SE will also trigger snapshot recovery.
> 
> * IronSide SE now re-requests snapshot recovery during boot if SDROM reports that snapshot recovery failed. (NCSDK-38309)

Reviewer: ```suggestion
* IronSide SE resilience against MRAM corruption.
  Secure domain CPU bus faults now trigger snapshot recovery.
  Corruption in other file systems used by IronSide SE will also trigger snapshot recovery.
  (NCSDK-37077)

* IronSide SE now re-requests snapshot recovery during boot if SDROM reports that snapshot recovery failed. (NCSDK-38309)
```

Fixed
-----

> * Adaptive body biasing (ABB) is now enabled before the system frequency is increased. (NCSDK-39151)
>   Operating at high frequencies without ABB violated the system specifications at extreme temperature corners and could lead to resets during boot.

Reviewer: ```suggestion
* Adaptive body biasing (ABB) is now enabled before the system frequency is increased.
  Operating at high frequencies without ABB violated the system specifications at extreme temperature corners and could have lead to resets during boot.
  (NCSDK-39151)
```
What is the issue that is being fixed here? This sounds like a change aimed at being compliant with system spec, so more an entry for Updated.

> * The Global Resource Control (GRC) active power domain is now forced to remain always on.
>   This incurs a negligible increase in current consumption, but is expected to fix a corner case issue where the device would hang until reset.

Reviewer: ```suggestion
* Rare issue where the device would hang until reset.
  The Global Resource Control (GRC) active power domain is now forced to remain always on.
  This incurs a negligible increase in current consumption.
```
"expected to fix" does not inspire confidence -- should this maybe go to Updated, given that this is just a change that attempts a fix, not a 100% fix?

> * Renamed IronSide Secure Element to IronSide Secure Enclave, aligning with the standard terminology.

Reviewer: How is this an issue to be fixed? Should it go to Updated?

IronSide Secure Enclave (IronSide SE) v23.6.0+29
================================================

Added
-----

* Support for ``UICR.SNAPSHOT_REGIONS`` to configure user-defined snapshot regions.
  This is only relevant when snapshot is enabled and configured in OTP.
* L2CACHE can now be disabled with PERIPHCONF to mitigate HMPAN-234 and HMPAN-242.

>   The L2CACHE is enabled by default. (NCSDK-38144)

Reviewer: Sounds like an entry for Updated.

Updated
-------

> * The secure domain clock frequency now matches the application domain clock frequency.

Reviewer: ```suggestion
* The Secure Domain clock frequency now matches the application domain clock frequency.
```
Capitalized in glossary: https://nrfconnectdocs.nordicsemi.com/ncs/latest/nrf/glossary.html#term-Secure-domain-SecDom

  Previously the secure domain always ran at 64 MHz, but it now matches the application domain frequency, which is at least 64 MHz and usually much higher.
  This change is expected to significantly improve the IronSide SE CPU and cryptographic accelerator performance. (NCSDK-38284)

> * CRACEN ECC operation countermeasures are now enabled (``CONFIG_CRACEN_ECC_COUNTERMEASURES=y``). (NCSDK-38269)

Reviewer: ```suggestion
* CRACEN ECC operation countermeasures are now enabled (``CONFIG_CRACEN_ECC_COUNTERMEASURES=y``.) (NCSDK-38269)
```
